app/controllers/resource_controller.rb

```diff
@@ -0,0 +1,138 @@
+class ResourceController < ApplicationController
+
+  helper_method :render_resource
+
+  def check_graph
+    onto_path = Rails.root.join('../101web/data/dumps/ontology.ttl')
+
+    if (!File.exists?(onto_path))
+      @result = nil
+      render file: 'resource/search.html.erb'
+      false
+    else
+      true
+    end
+  end
+
+  def landing
+
+    if(check_graph)
+      host = request.host
+      port = ':'+request.port.to_s
+
+      # + prepare subject ----------------------
+      @subject = RDF::URI.new(scheme: request.scheme.dup,
+                              authority: host + port,
+                              host: host,
+                              port: request.port,
+                              path: 'resource/101companies')
+      # - prepare subject ----------------------
+
+      # + execute rdf querys -----------------
+      respond_to do |format|
+        format.json { render :json => [] }
+        format.xml  { render :xml  => [].to_rdfxml }
+        format.html {
+          # + queries on graph -------------------
+          @result = $graph.query([:s, :p, @subject]).to_a.uniq{|s,p,o| s.pname}.sort_by { |s,p,o| s.pname }
+          render file: 'resource/landing.html.erb'
+        }
+      end
+    end
+  end
+
+  def get
+    if(check_graph)
+      host = request.host
+      port = ':'+request.port.to_s
+
+      # + prepare subject ----------------------
+      @subject = RDF::URI.new(scheme: request.scheme.dup,
+                             authority: host + port,
+                             host: host,
+                             port: request.port,
+                             path: 'resource/' + params[:resource_name])
+      # - prepare subject ----------------------
+
+      sub_set = $graph.query([@subject, :p, :o]).to_set
+      obj_set = $graph.query([:s, :p, @subject]).to_set
```

Comment:
After the first query, the $graph might update its internal object and we query from 2 different graphs. I think we need some kind of transactional process here.